3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/UserAgentContextEnrichment.scala

```diff
@@ -0,0 +1,157 @@
+ * browser resolution, etc.
+ */
+object UserAgentContextEnrichment {
+  
+  /**
+   * The Tracker Protocol's pattern
+   * for a screen resolution - for
+   * details see:
+   *
+   * https://github.com/snowplow/snowplow/wiki/snowplow-tracker-protocol#wiki-browserandos
+   */
+  private val ResRegex = """(\d+)x(\d+)""".r
+
+  /**
+   * Case class to wrap everything we
+   * can extract from the useragent
+   * using UserAgentUtils.
+   *
+   * TODO: update this definition when
+   * we swap out UserAgentUtils for
+   * ua-parser
+   */
+  case class ClientAttributes(
+      // Browser
+      //browserName: String,
+      browserFamily: String,
+      browserMinor: String,
+      browserMajor: String,
+      //browserVersion: Option[String],
+      //browserType: String,
+      //browserRenderEngine: String,
+      // OS the browser is running on
+      //osName: String,
+      osFamily: String,
+      osMinor: String,
+      osMajor: String,
+      osManufacturer: String,
+      // Hardware the OS is running on
+      deviceType: String,
+      deviceIsMobile: Boolean)
+
+  /**
+   * Extracts view dimensions (e.g. screen resolution,
+   * browser/app viewport) stored as per the Tracker
+   * Protocol:
+   *
+   * https://github.com/snowplow/snowplow/wiki/snowplow-tracker-protocol#wiki-browserandos
+   *
+   * @param field The name of the field
+   *        holding the screen dimensions
+   * @param res The packed string
+   *        holding the screen dimensions
+   * @return the ResolutionTuple or an
+   *         error message, boxed in a
+   *         Scalaz Validation
+   */
+  val Pattern = "(iPhone|webOS|iPod|Android|BlackBerry|mobile|SAMSUNG|IEMobile|OperaMobi|Nokia)".r.unanchored
+
+  def isMobile[A](implicit request: Request[A]): Boolean = {
+    request.headers.get("User-Agent").exists(agent => {
+      agent match {
+        case Pattern(a) => true
+        case _ => false
+      }
+    })
+  }
+
+  val extractViewDimensions: (String, String) => Validation[String, ViewDimensionsTuple] = (field, res) =>
+    res match {
+      case ResRegex(width, height) =>
+        try {
+          (width.toInt: JInteger, height.toInt: JInteger).success
+        } catch {
+          case _ => "Field [%s]: view dimensions [%s] exceed Integer's max range".format(field, res).fail
+        }
+      case _ => "Field [%s]: [%s] does not contain valid view dimensions".format(field, res).fail
+    }
+
+  /**
+   * Extracts the client attributes
+   * from a useragent string, using
+   * uap-java Library.
+   *
+   *
+   * @param useragent The useragent
+   *        String to extract from.
+   *        Should be encoded (i.e.
+   *        not previously decoded).
+   * @return the ClientAttributes or
+   *         the message of the
+   *         exception, boxed in a
+   *         Scalaz Validation
+   */
+  def extractClientAttributes(useragent: String): Validation[String, ClientAttributes] = 
+
+    try {
+      Parser uaParser = new Parser()
```

Comment:
This is Java code. There's no way this is compiling. Don't open PRs that don't compile.